### ISSUE TITLE
Running CogVideoX-5B on T4/V100 Free Colab Space

### ISSUE THREAD

**Issue by ProKSMT** [THUDM/CogVideo#204]
torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 56.50 GiB.

V100 32G

5B model, `enable_model_cpu_offload()` option and `pipe.vae.enable_tiling()` optimization were enabled

using diffusers (cli_demo.py)

**Comment by zRzRzRzRzRzRzR** [comment]
update diffusers to 0.30.1

**Comment by ProKSMT** [comment]
I am using diffusers 0.30.1

**Comment by zRzRzRzRzRzRzR** [comment]
can you try the code in `cogvideox-dev`breanch with it requirement and try again with cli_demo.py 
also, use `breakpoint()` to locate the OOM code line,thks

**Comment by ProKSMT** [comment]
I don‘t know how to use that to locate the OOM code line. Maybe this log will be helpful.
![image](https://github.com/user-attachments/assets/9ace2051-fa99-4db5-b5c6-4ced88f78cbd)

And I will test the `cogvideox-dev` branch as soon as possible.

**Comment by GuanleiGao** [comment]
I had the same problem with V100, and it was solved by switching to A10. It seems to be a graphics card problem

**Comment by ProKSMT** [comment]
I think so too. I found that V100 does not support bf16. I switched the dtype to fp16 and it worked (main branch).
So I think it might not be necessary to test on the dev branch.
However, I don't know exactly how the V100 leads to OOM just because it doesn't support bf16. Maybe the auto type conversion make the VRAM consumption multiply I guess.

**Comment by Exploder98** [comment]
I'm seeing this on my AMD RX 6900 XT. Changing the dtype does not have any effect, though. Could this have something to do with Flash Attention or Memory efficient attention support? I know that on my GPU neither of those work.

**Comment by zRzRzRzRzRzRzR** [comment]
I think we need to try this issue. The 3060 desktop version has only 12G, but it can run the 5B model normally. However, there is feedback from developers that the V100 32G has problems running the 5B model, while the 2B model runs normally. I will check if it is a precision issue.

**Comment by ProKSMT** [comment]
OK, I just tested it on the dev branch, and the same issue occurred. It also shows as 56.50G
![image](https://github.com/user-attachments/assets/fbb7b534-7d67-40b7-84fc-252e5178cb5f)

**Comment by zRzRzRzRzRzRzR** [comment]
Check if several key positions are open
1. Do not attempt to enable online quantization, this may cause errors on the GPU in this architecture.

2.  Try to check several key nodes
```
pipe = CogVideoXPipeline.from_pretrained("THUDM/CogVideoX-5b", torch_dtype=torch.float16)
pipe = CogVideoXPipeline.from_pretrained(
"THUDM/CogVideoX-5b"
text_encoder=text_encoder,
transformer=transformer
woe=woe,
torch_dtype=torch.float16
)
```

You must use FP16 on T4 unless you are using a GPU with Ampere or higher architecture that supports BF16
Additionally, do not use .to(device), as this allows for better compression on the CPU and memory, rather than transferring the entire complete model to the GPU.

3. Finally, check whether these four memory-saving schemes are enabled
```
pipe.enable_model_cpu_offload()
pipe.enable_sequential_cpu_offload()
pipe.vae.enable_slicing()
pipe.vae.enable_tiling()
```
I am already running normally on T4 on colab
<img width="814" alt="image" src="https://github.com/user-attachments/assets/4feffdaa-aa18-4932-b784-0ca726ea8ff7">

Please check if this can help you

**Comment by ProKSMT** [comment]
So it seems that the V100 can not run in BF16 mode. But It looks like that FP16 mode is not as good as BF16.
![image-20240830143711249](https://github.com/user-attachments/assets/6a4d97f2-17ca-4196-9932-d91cafcdcd14)

Will you release a special FP16 version of the 5B model ?

**Comment by zRzRzRzRzRzRzR** [comment]
> So it seems that the V100 can not run in BF16 mode. But It looks like that FP16 mode is not as good as BF16. ![image-20240830143711249](https://private-user-images.githubusercontent.com/177197291/363068545-6a4d97f2-17ca-4196-9932-d91cafcdcd14.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwNzIxNzksIm5iZiI6MTcyNTA3MTg3OSwicGF0aCI6Ii8xNzcxOTcyOTEvMzYzMDY4NTQ1LTZhNGQ5N2YyLTE3Y2EtNDE5Ni05OTMyLWQ5MWNhZmNkY2QxNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgzMVQwMjM3NTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xNmI5NDQ2M2Y4OTBiYzM1NjNmMDIwMDdkNmU2ZDNkMGM4MmE1NzhkODUyY2IwYTJlMzk4MjI5MzExNmI4MGZjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.sYRaDA4IFDH2EzAvQaAQbdysDGvSt5k1hS4dd7xBijc)
> 
> Will you release a special FP16 version of the 5B model ?

We tried, but the results weren’t ideal. The 5B model is currently recommended to run at BF16 precision, which is also the precision we used for training. Converting to FP16 leads to suboptimal performance. However, the 2B model has lower compatibility requirements and can run effectively in FP16.

**Comment by camenduru** [comment]
free colab: https://github.com/camenduru/CogVideoX-5B-jupyter

**Comment by lonngxiang** [comment]
use  FP16 on T4 ,still error
https://colab.research.google.com/drive/14TTaDTM3_lk69qKb5u4-1_gm_YK6lM3m?usp=sharing

```
# Create pipeline and run inference
pipe = CogVideoXPipeline.from_pretrained(
    "THUDM/CogVideoX-5b",
    text_encoder=text_encoder,
    transformer=transformer,
    vae=vae,
    torch_dtype=torch.float16,
)
pipe.enable_model_cpu_offload()
# pipe.enable_sequential_cpu_offload()
pipe.vae.enable_slicing()
pipe.vae.enable_tiling()

    
```
![image](https://github.com/user-attachments/assets/35005620-b30c-4283-a092-d015e18718df)

**Comment by lonngxiang** [comment]
pipe.enable_sequential_cpu_offload()  not work

**Comment by zRzRzRzRzRzRzR** [comment]
why not work ,this should use with diffusers>=0.30.1 and using FP16 model,not INT8

**Comment by lonngxiang** [comment]
> why not work ,this should use with diffusers>=0.30.1 and using FP16 model,not INT8
i use diffusers 0.30.2；and not int8, you can see my code this link:https://colab.research.google.com/drive/14TTaDTM3_lk69qKb5u4-1_gm_YK6lM3m?usp=sharing

**Comment by zRzRzRzRzRzRzR** [comment]
<img width="1205" alt="image" src="https://github.com/user-attachments/assets/c7a5f0b2-6bbd-4928-8bcb-f0dcfef39a85">
This is not right and we have upload a colab example in our readme

**Comment by lonngxiang** [comment]
where is colab link? i can't see that, please send down, T4 can run?

**Comment by zRzRzRzRzRzRzR** [comment]
https://github.com/camenduru/CogVideoX-5B-jupyter

**Comment by lonngxiang** [comment]
> Member
这生成时间需要1个多小时吗
![image](https://github.com/user-attachments/assets/b9efb99f-1265-4bdc-8792-2b92ec60272f)

**Comment by zRzRzRzRzRzRzR** [comment]
There is no need, but it does take a long time (in my T4 colab similar code it takes about 20 minutes, this is due to the computational power limitations of this generation of GPUs, and to compress memory usage, a lot of time-for-space solutions are used, resulting in very slow speed. Additionally, the T4 cannot run BF16 models, and the quality of FP16 inference cannot be guaranteed, so we recommend using newer GPUs for inference and fine-tuning.

**Comment by lonngxiang** [comment]
takes 1hour 
![image](https://github.com/user-attachments/assets/c5788d4e-253a-416d-ac24-983cbc7772e0)

**Comment by e7470** [comment]
Hi i was guided to post here. I am receiving an error while generating video. The processing start, steps start to go on very quickly and then ...ERROR.. I have tried

    switch both models 2B/ 5B
    reduce the steps and guidance etc
    switch float types

My PC
Dell T7810 with dual E-2643 v4 CPUs 64Gb DDR4 ram and dual P6000 GPUs, 24 GB each, total 48Gb. Ubuntu 24.04 LTS Desktop

I am not sure if its a memory issue, I thought this should work with one 24GB card as per the BIO of the app.
I am using pinokoi and got this last error
File "/home/mandy/pinokio/api/cogvideo.git/app/env/lib/python3.10/site-packages/diffusers/models/attention_processor.py", line 1934, in call
hidden_states = F.scaled_dot_product_attention(
torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 35.31 GiB. GPU
